qiskit/aqua/components/oracles/deutsch_jozsa_oracle.py

```diff
@@ -38,73 +40,68 @@ class DeutschJozsaOracle(Oracle):
             'type': 'object',
             'properties': {
                 'bitmap': {
-                    "type": ["object", "null"],
+                    "type": ["object"],
                 }
             },
             'additionalProperties': False
         }
     }
 
-    def __init__(self, bitmap):
+    def __init__(self, bitmap, mct_mode='basic'):
```

Comment:
> Instead of bitmap (a bitstring of length 2**n), can we provide a DNF formula instead as input to the oracle?

here I didn't change @attp 's original design of the use of bitmap dictionary as the input format; I suppose we can support multiple input formats

Reply:
EXOR sum of products (EXOR polynomial) can be a better alternative to the disjunctive sum of products due to a combination of: 
1. often smaller number of terms in EXOR polynomial compared to the number of terms in DNF (including a reduced disjunction of conjunctions; famously x_1 \oplus x_2 \oplus ... \oplus x_n)
2. ease of implementation as a quantum (reversible) circuit due to the availability of multiple-control Toffoli gates.
Input description may furthermore admit arbitrary formulas over Boolean logic.  Mapping it to quantum circuits can then be done by individually decomposing binary operations used into 2x2 and 3x3 reversible gates.